js_modules/dagster-ui/packages/ui-core/src/runs/CellTruncationProvider.tsx

```diff
@@ -22,97 +22,86 @@ const OverflowButtonContainer = styled.div`
 
-  onView = () => {
-    const {onExpand} = this.props;
-    if (onExpand) {
-      onExpand();
+  const dialogContents = () => {
+    const message = contentContainerRef.current?.textContent;
+    return message ? <div style={{whiteSpace: 'pre-wrap'}}>{message}</div> : null;
+  };
+
+  const onView = () => {
+    if (props.onExpand) {
+      props.onExpand();
     } else {
-      this.setState({showDialog: true});
+      setState((prev) => ({...prev, showDialog: true}));
     }
   };
 
-  render() {
-    const style = {...this.props.style, overflow: 'hidden'};
+  const style = {...props.style, overflow: 'hidden'};
 
-    return (
-      <div style={style}>
-        <div ref={this.contentContainerRef}>{this.props.children}</div>
-        {(this.state.isOverflowing || this.props.forceExpandability) && (
-          <>
-            <OverflowFade />
-            <OverflowButtonContainer>
-              <Button intent="primary" icon={<Icon name="unfold_more" />} onClick={this.onView}>
-                View full message
-              </Button>
-            </OverflowButtonContainer>
-            {this.props.onExpand ? null : (
-              <Dialog
-                canEscapeKeyClose
-                canOutsideClickClose
-                isOpen={this.state.showDialog}
-                onClose={() => this.setState({showDialog: false})}
-                style={{width: 'auto', maxWidth: '80vw'}}
-              >
-                <div>{this.dialogContents()}</div>
-                <DialogFooter topBorder>
-                  <Button intent="primary" onClick={() => this.setState({showDialog: false})}>
-                    Done
-                  </Button>
-                </DialogFooter>
-              </Dialog>
-            )}
-          </>
-        )}
-      </div>
-    );
-  }
-}
+  return (
+    <div style={style}>
+      <div ref={contentContainerRef}>{props.children}</div>
+      {(state.isOverflowing || props.forceExpandability) && (
+        <>
+          <OverflowFade />
+          <OverflowButtonContainer>
+            <Button intent="primary" icon={<Icon name="unfold_more" />} onClick={onView}>
+              View full message
+            </Button>
+            {props.buttons}
+          </OverflowButtonContainer>
+          {props.onExpand ? null : (
+            <Dialog
+              canEscapeKeyClose
+              canOutsideClickClose
+              isOpen={state.showDialog}
+              onClose={() => setState((prev) => ({...prev, showDialog: false}))}
+              style={{width: 'auto', maxWidth: '80vw'}}
+            >
+              <div>{dialogContents()}</div>
+              <DialogFooter topBorder>
+                <Button
+                  intent="primary"
+                  onClick={() => setState((prev) => ({...prev, showDialog: false}))}
+                >
+                  Done
+                </Button>
+              </DialogFooter>
+            </Dialog>
+          )}
+        </>
+      )}
+    </div>
+  );
+};
```

Comment:
Is the class->function refactor worth doing in a separate followup PR? I could imagine there being subtle bugs here that might make it worthwhile to keep a revert simple without affecting the rest of the changes.